uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/FederatedLoginIT.java

```diff
@@ -0,0 +1,168 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.oauth2.common.util.RandomValueStringGenerator;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.web.client.RestTemplate;
+import java.net.Inet4Address;
+import java.net.UnknownHostException;
+import java.util.Arrays;
+import static org.junit.Assert.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = DefaultIntegrationTestConfig.class)
+public class FederatedLoginIT extends BaseSamlTestIT {
+
+    public static final String IDP_ENTITY_ID = "cloudfoundry-saml-login";
+
+    private final SamlTestUtils samlTestUtils = new SamlTestUtils();
+
+    @Autowired
+    WebDriver webDriver;
+
+    @Value("${integration.test.base_url}")
+    String baseUrl;
+
+
+    ServerRunning serverRunning = ServerRunning.isRunning();
+
+
+    @Before
+    public void clearWebDriverOfCookies() throws Exception {
+        samlTestUtils.initialize();
+        webDriver.get(baseUrl + "/logout.do");
+        webDriver.manage().deleteAllCookies();
+        webDriver.get(baseUrl.replace("localhost", "testzone1.localhost") + "/logout.do");
+        webDriver.manage().deleteAllCookies();
+        webDriver.get(baseUrl.replace("localhost", "testzone2.localhost") + "/logout.do");
+        webDriver.manage().deleteAllCookies();
+        assertTrue("Expected testzone1.localhost and testzone2.localhost to resolve to 127.0.0.1", doesSupportZoneDNS());
+    }
+
+    @Before
+    public void setup() {
+        String token = IntegrationTestUtils.getClientCredentialsToken(baseUrl, "admin", "adminsecret");
+
+        ScimGroup group = new ScimGroup(null, "zones.testzone1.admin", null);
+        IntegrationTestUtils.createGroup(token, "", baseUrl, group);
+
+        group = new ScimGroup(null, "zones.testzone2.admin", null);
+        IntegrationTestUtils.createGroup(token, "", baseUrl, group);
+
+        group = new ScimGroup(null, "zones.uaa.admin", null);
+        IntegrationTestUtils.createGroup(token, "", baseUrl, group);
+    }
+
+    protected boolean doesSupportZoneDNS() {
+        try {
+            return Arrays.equals(Inet4Address.getByName("testzone1.localhost").getAddress(),
+                    new byte[]{127, 0, 0, 1})
+                    && Arrays.equals(Inet4Address.getByName("testzone2.localhost").getAddress(),
+                    new byte[]{127, 0, 0, 1});
+        } catch (UnknownHostException e) {
+            return false;
+        }
+    }
+
+    /**
+     * In this test testzone1 acts as the SAML IdP and testzone2 acts as the SAML SP.
+     */
```

Comment:
This test is using testzone1 that is provisioned using orchestrator zone API; zone id is randomly generated by UAA service upon zone creation. However, some other ITs (SamlLoginWithLocalIdpIT, SamlLoginIT, TotpMfaEndpointIntegrationTests, etc) are also using testzone1 that is provisioned using native zone APIs.  These tests assume that zone id is "testzone1." Since tests are executed in parallel, changing testzone1 id makes tests to fail.

Change testzone1 to saml-idp-zone nd testzone2 to saml-sp-zone. You will have to add saml-idp-zone.localhost and saml-sp-zone.localhost to [Jenkinsfile#L219](https://github.com/GESoftware-CF/uaa/blob/orchestrator_intergration_test/Jenkinsfile#L219) as well so they are mapped to localhost in Jenkins.

Reply:
Done saml-idp-zone and saml-sp-zone were giving 500 error so added it as samlidpzone and samlspzone